### DISCUSSION TITLE
Creeper

### DISCUSSION THREAD

**Discussion by deftesthawk** [sf-inc/sihywtcamd#21] [Ideas]
When a creeper explodes it could release a configurable random amount of baby creepers, this would incentivize people who play with creeper damage turned off to not just let them explode. a separate option to turn both types of Creepers world damage off would be nice as well, as I personally find filling in world holes more annoying than it adding actual difficulty to the game.

Or when they explode on you it will make your sound go muffled, give you blindness and slowness for a few seconds

**Comment by charlyb01** [comment]
I don't think baby creepers is vanilla friendly. Do you imagine baby creepers in the game? I don't. If you don't want creepers to make holes, you can turn off "mobGriefing" gamerule.
I will think a bit more about the idea of being deaf/blind after an explosion (not only creepers one) because it makes sense to me. Thanks for the idea!

**Reply by deftesthawk** [reply]
Yeah true the only way around the baby creeper idea would be to make it so they didn’t do world damage, just a small non world damaging explosion

**Reply by deftesthawk** [reply]
They definitely need something to increase challenge hopefully the explosion giving players those negative effects will balance it out a bit more

**Reply by deftesthawk** [reply]
Maybe they could move slightly faster when getting close or something, or add some dodge mechanics so it’s harder to keep them away

**Reply by charlyb01** [reply]
Creepers now deal blindness and tinnitus effects. Their duration depends on how close you are from the creeper explosion. The tinnitus effect is a new effect that decrease every sound that a player can hear, except ambient and weather ones that are increased, to simulate (roughly) a tinnitus. The effect will be stronger depending on how close you are, once again.
What do you think about it? I'll let you try this too =)
[sihywtcamd-1.6-beta2.zip](https://github.com/sf-inc/sihywtcamd/files/7027480/sihywtcamd-1.6-beta2.zip)

**Comment by GalatynF** [comment]
There is a mod similar to your first idea by the way : https://www.curseforge.com/minecraft/mc-mods/creeper-spores
I really like it, since I hate creeper holes too.

**Comment by charlyb01** [comment]
In a stupid mod that I made in private, I did this: creepers don't die from damage explosions. Instead, they ignite. They will still die in their own explosion, but if another creeper is close, it will start exploding too. What do you think about this stupid thing? x)

**Reply by deftesthawk** [reply]
That sounds pretty awesome

**Reply by deftesthawk** [reply]
I tested out the beta2 and here are my notes

- I'm not sure about the blindness effect as it feels a bit too much, maybe it could be changed to a strong blurring effect instead
- The audio didn't seem to cut out for me when a creeper exploded
- I think the duration should be increased especially if it's changed to a blur effect instead of blind

**Reply by charlyb01** [reply]
* Its explosion deal blindness and tinnitus effects, duration depending on your distance with the creeper
* Tinnitus effect **reduce** every sound and **increase** ambient and weather sounds to simulate a tinnitus. The 
  multiplier depends on the amplifier of the effect (from 0 to 99, for percentages)
* At the maximum, sounds will be reduced at 50% their original values or increased of 50%. (ex: music 50% and ambient 60% become music 25% and ambient 90%)
* When a creeper gives you tinnitus, the amplifier depends also on the distance

It may be a bit more clear now why your sound didn't seem to cut out. I'm open to suggesions about the behavior of the new effect ;)
Blurring effect is a cool idea but seems a bit annoying to implement

**Comment by charlyb01** [comment]
This was implemented and modified a while ago, see [Changelog](https://github.com/sf-inc/sihywtcamd/blob/master/CHANGELOG.md#v16). It now deals Weakness and Mining Fatigue.  
Feel free to reopen discussion if needed =)
